python/pyspark/ml/tests/test_pipeline.py

```diff
@@ -58,6 +70,109 @@ def doTransform(pipeline):
         # check that failure to set stages param will raise KeyError for missing param
         self.assertRaises(KeyError, lambda: doTransform(Pipeline()))
 
+    def test_classification_pipeline(self):
+        df = self.spark.createDataFrame([(1, 1, 0, 3), (0, 2, 0, 1)], ["y", "a", "b", "c"])
+
+        assembler = VectorAssembler(outputCol="features")
+        assembler.setInputCols(["a", "b", "c"])
+
+        scaler = MaxAbsScaler(inputCol="features", outputCol="scaled_features")
+
+        lr = LogisticRegression(featuresCol="scaled_features", labelCol="y")
+        lr.setMaxIter(1)
+
+        pipeline = Pipeline(stages=[assembler, scaler, lr])
+        self.assertEqual(len(pipeline.getStages()), 3)
+
+        # Pipeline save & load
+        with tempfile.TemporaryDirectory(prefix="classification_pipeline") as d:
+            pipeline.write().overwrite().save(d)
```

Comment:
I made a PR (https://github.com/apache/spark/pull/49610 ) to combine the read/write for estimator and model in the same directory, which could avoid re-creating the directory twice, and the "overwrite" could be exactly tested.

Reply:
sounds good!

Reply:
since `overwrite` is not supported for now, we will need to create sub dictionary